v2/yarpcprotobuf/protoc-gen-yarpc-go/integration_test.go

```diff
@@ -0,0 +1,126 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package main
+
+import (
+	"context"
+	"net/url"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	yarpc "go.uber.org/yarpc/v2"
+	"go.uber.org/yarpc/v2/yarpchttp"
+	commonpb "go.uber.org/yarpc/v2/yarpcprotobuf/protoc-gen-yarpc-go/internal/tests/gen/proto/src/common"
+	keyvaluepb "go.uber.org/yarpc/v2/yarpcprotobuf/protoc-gen-yarpc-go/internal/tests/gen/proto/src/keyvalue"
+	streampb "go.uber.org/yarpc/v2/yarpcprotobuf/protoc-gen-yarpc-go/internal/tests/gen/proto/src/stream"
+	keyvalue "go.uber.org/yarpc/v2/yarpcprotobuf/protoc-gen-yarpc-go/internal/tests/src/keyvalue"
+	stream "go.uber.org/yarpc/v2/yarpcprotobuf/protoc-gen-yarpc-go/internal/tests/src/stream"
+	"go.uber.org/yarpc/v2/yarpcrouter"
+)
+
+func newKeyValueClient(t *testing.T) keyvaluepb.StoreYARPCClient {
+	dialer := &yarpchttp.Dialer{}
+	require.NoError(t, dialer.Start(context.Background()))
+
+	outbound := &yarpchttp.Outbound{
+		URL:    &url.URL{Scheme: "http", Host: "127.0.0.1:8888"},
+		Dialer: dialer,
+	}
+
+	return keyvaluepb.NewStoreYARPCClient(yarpc.Client{
+		Caller:  "test",
+		Service: "keyvalue",
+		Unary:   outbound,
+	})
+}
+
+func newHelloClient(t *testing.T) streampb.HelloYARPCClient {
+	return streampb.NewHelloYARPCClient(yarpc.Client{
+		Caller:  "test",
+		Service: "hello",
+	})
+}
+
+func startProcedures(t *testing.T, service string, procedures []yarpc.Procedure) (stop func() error) {
+	router := yarpcrouter.NewMapRouter(service)
+	router.Register(procedures)
+
+	inbound := &yarpchttp.Inbound{
+		Addr:   ":8888",
+		Router: router,
+	}
+	require.NoError(t, inbound.Start(context.Background()))
+
+	return func() error { return inbound.Stop(context.Background()) }
+}
+
+func TestIntegration(t *testing.T) {
```

Comment:
optional, you may want to put these in independent functions since they don't share any logic, eg
`TestUnaryIntegration`, `TestBidiIntegration`

Reply:
I'll leave this as is for now. I initially wrote them as sub-tests so that they could access a shared scope. If the tests end up not referencing each other after we introduce the gRPC transport layer and finish the stream test, then I can break these out into separate tests.